### PR TITLE
Add Typescript language module

### PR DESCRIPTION
**Pull request by Kr0nox**
This PR adds a language module for parsing TypeScript and JavaScript.
Since the grammar specifies no version, but code from TS v5 can be parsed (tho no tokens are extracted) the version is estimated to be v5.

The token extraction is inspired by Java and Python

### CODE REVIEW

File: languages/typescript/src/main/java/de/jplag/typescript/grammar/TypeScriptLexerBase.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import java.util.Deque;
+
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.Token;
+
+/**
+ * Copied from https://github.com/antlr/grammars-v4/tree/master/javascript/typescript/Java
+ */
+abstract class TypeScriptLexerBase extends Lexer {
```
Comment (dfuchss) [inline languages/typescript/src/main/java/de/jplag/typescript/grammar/TypeScriptLexerBase.java:13]:
```suggestion
@SuppressWarnings({"java:S100", "java:S5993", "java:S1155", "java:S6208"})
abstract class TypeScriptLexerBase extends Lexer {
```

I'd suppress warnings of sonarlint for the copied classes. Not nice, but I don't think we want to let these external dependencies count as issue . (but not sure)

File: README.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ In the following, a list of all supported languages with their supported languag
 | [EMF Metamodel](https://www.eclipse.org/modeling/emf/) |  2.25.0 | emf               |                                beta                                 |    EMF    |
 | [EMF Model](https://www.eclipse.org/modeling/emf/)     |  2.25.0 | emf-model         |                                alpha                                |    EMF    |
 | [LLVM IR](https://llvm.org)                            |      15 | llvmir            |                                beta                                 |  ANTLR 4  |
+| [TypeScript](https://www.typescriptlang.org/)          |      ~5 | typescript        |                                beta                                 |  ANTLR 4  |
```
Comment (sebinside) [inline README.md:39]: Add a hint for javascript support and maybe add a link at the ~5 to the ANTLR readme that discusses what is supported?

File: languages/typescript/src/main/java/de/jplag/typescript/grammar/TypeScriptLexerBase.java (hunk 1 above)
Comment (Kr0nox) [reply languages/typescript/src/main/java/de/jplag/typescript/grammar/TypeScriptLexerBase.java:14]: We decided on modifying the files from the ANTRL Repo to fit the JPlag code style

File: languages/typescript/pom.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>org.antlr</groupId>
+            <artifactId>antlr4-runtime</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>de.jplag</groupId>
+            <artifactId>language-antlr-utils</artifactId>
+            <version>4.4.0-SNAPSHOT</version>
```
Comment (dfuchss) [inline languages/typescript/pom.xml:20]: Use revision or dependency management here

**Comment by Kr0nox** [comment]
@sebinside wanted to have a look at the choice of tokens. I dont know if he did that already

**Pull request merged**